sudo make install fails on FreeBSD

Hi, there are several issues that I have run into which prevent me from building from source. I am using FreeBSD 11 with clang 3.8. I also tried using gcc6 but ran into similar issues.

```CMake
[  1%] Built target xpp
Scanning dependencies of target polybar
[  2%] Building CXX object bin/CMakeFiles/polybar.dir/adapters/alsa.cpp.o
In file included from <built-in>:350:
<command line>:1:9: error: macro name must be an identifier
#define -I/usr/local/include 1
        ^
In file included from /usr/home/tim/polybar/src/adapters/alsa.cpp:1:
/usr/home/tim/polybar/include/adapters/alsa.hpp:4:10: fatal error: 'endian.h' file not found
#include <endian.h>
         ^
2 errors generated.
*** Error code 1

Stop.
make[2]: stopped in /usr/home/tim/polybar/build
*** Error code 1

Stop.
make[1]: stopped in /usr/home/tim/polybar/build
*** Error code 1

Stop.
make: stopped in /usr/home/tim/polybar/build
```

I don't know how to fix the first error, but to fix the second, I change `#include <endian.h>`  to `#include <sys/endian.h>` in `include/adapters/alsa.hpp`

After making that change, the build output becomes:

```CMake
[  1%] Built target xpp
Scanning dependencies of target polybar
[  2%] Building CXX object bin/CMakeFiles/polybar.dir/adapters/alsa.cpp.o
In file included from <built-in>:350:
<command line>:1:9: error: macro name must be an identifier
#define -I/usr/local/include 1
        ^
In file included from /usr/home/tim/polybar/src/adapters/alsa.cpp:1:
In file included from /usr/home/tim/polybar/include/adapters/alsa.hpp:26:
/usr/local/include/alsa/pcm.h:597:19: error: zero size arrays are an extension
      [-Werror,-Wzero-length-array]
        unsigned int pos[0];    /**< channel position array */
                         ^
/usr/home/tim/polybar/src/adapters/alsa.cpp:247:16: error: use of undeclared identifier 'pow10'
  normalized = pow10((vol_total / chan_n - vol_max) / 6000.0);
               ^
/usr/home/tim/polybar/src/adapters/alsa.cpp:249:16: error: use of undeclared identifier 'pow10'
    min_norm = pow10((vol_min - vol_max) / 6000.0);
               ^
/usr/home/tim/polybar/src/adapters/alsa.cpp:296:16: error: use of undeclared identifier 'pow10'
    min_norm = pow10((vol_min - vol_max) / 6000.0);
               ^
5 errors generated.
*** Error code 1

Stop.
make[2]: stopped in /usr/home/tim/polybar/build
*** Error code 1

Stop.
make[1]: stopped in /usr/home/tim/polybar/build
*** Error code 1

Stop.
make: stopped in /usr/home/tim/polybar/build
```

I've got to linking stage by doing the following:
Run `build.sh` and disable unsupported stuff (e.g. git as bsd has no `inotify_<method>` syscalls) and stuff you don't want:

```
** Setting build options
** Include support for "internal/i3" (requires i3) ------------------- [Y/n]: n
** Include support for "internal/volume" (requires alsalib) ---------- [Y/n]: y
** Include support for "internal/network" (requires wireless_tools) -- [Y/n]: n
** Include support for "internal/mpd" (requires libmpdclient) -------- [Y/n]: n
** Include support for "internal/github" (requires libcurl) ---------- [Y/n]: n
** Build "polybar-msg" used to send ipc messages --------------------- [Y/n]: y
```

Adding an `#ifdef _FreeBSD_` for `sys/endian.h` statement.

Disabling `-Werror` and pedantic errors, I guess the author uses gcc and clang has different/additional warnings and errors.

Add the `_WITH_DPRINTF` define for dprintf, see https://reviews.freebsd.org/D6302 (I imagine this won't be an issue with FreeBSD 12.0-CURRENT.

----
Despite these changes though, and despite explicitly disabling `git` in the build.sh prompt, my build fails at linking:
CMakeFiles/polybar.dir/utils/inotify.cpp.o: In function `polybar::v3_0_4_git::inotify_watch::~inotify_watch()':
/home/ben/polybar/src/utils/inotify.cpp:19: undefined reference to `inotify_rm_watch'
CMakeFiles/polybar.dir/utils/inotify.cpp.o: In function `polybar::v3_0_4_git::inotify_watch::attach(int)':
/home/ben/polybar/src/utils/inotify.cpp:30: undefined reference to `inotify_init'
/home/ben/polybar/src/utils/inotify.cpp:33: undefined reference to `inotify_add_watch'
CMakeFiles/polybar.dir/utils/inotify.cpp.o: In function `polybar::v3_0_4_git::inotify_watch::remove(bool)':
/home/ben/polybar/src/utils/inotify.cpp:43: undefined reference to `inotify_rm_watch'
clang++: error: linker command failed with exit code 1 (use -v to see invocation)

I'm not really interested in forking this and maintaining a build for FreeBSD, hopefully this helps somebody who is.

--- a/cmake/01-core.cmake
+++ b/cmake/01-core.cmake
@@ -31,8 +31,17 @@ set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O2")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pedantic")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pedantic-errors")
 
+if (CMAKE_SYSTEM_NAME STREQUAL "FreeBSD")
+  # Need dprintf() for FreeBSD 11.1 and older
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -D_WITH_DPRINTF")
+
+  # libinotify uses c99 extension, so suppress this error
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-c99-extensions")
+endif()
+
 if(${CMAKE_CXX_COMPILER_ID} STREQUAL Clang)
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-error=parentheses-equality")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-zero-length-array")
 endif()
 
 set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -DDEBUG")

--- a/cmake/03-libs.cmake
+++ b/cmake/03-libs.cmake
@@ -21,3 +21,8 @@ querylib(WITH_XRENDER "pkg-config" xcb-render libs dirs)
 querylib(WITH_XRM "pkg-config" xcb-xrm libs dirs)
 querylib(WITH_XSYNC "pkg-config" xcb-sync libs dirs)
 querylib(WITH_XCURSOR "pkg-config" xcb-cursor libs dirs)
+
+# FreeBSD Support
+if(CMAKE_SYSTEM_NAME STREQUAL "FreeBSD")
+  querylib(TRUE "pkg-config" libinotify libs dirs)
+endif()

--- a/include/adapters/alsa/generic.hpp
+++ b/include/adapters/alsa/generic.hpp
@@ -4,7 +4,11 @@
 #include <alsa/asoundlib.h>
 #else
 #include <assert.h>
+
+#ifndef __FreeBSD__
 #include <endian.h>
+#endif
+
 #include <errno.h>
 #include <fcntl.h>
 #include <stdarg.h>

